MOxUnit/@MOxUnitTestOutcome/getSummaryStr.m

```diff
@@ -46,9 +35,57 @@
 
 
 function str=obj2xml(obj)
-    str=[];
-    % TODO
-    %
-    % Should use isNonFailure, isSuccess, getOutcomeStr and
-    % getSummaryContent
-    error('not implemented');
\ No newline at end of file
+    test_=getTest(obj);
+
+    prefix=sprintf('<testcase classname="%s" name="%s" time="%.3f"',...
+                moxunit_util_escape_xml(get_classname(test_)),...
+                moxunit_util_escape_xml(getName(test_)),...
+                getDuration(obj));
+
+    if isSuccess(obj)
+        infix='';
+        suffix=' />';
+    else
+        content=getSummaryContent(obj);
+
+        % outcome is 'failure', 'error', or 'skipped'
+        outcome_verbosity=2;
+        outcome=getOutcomeStr(obj, outcome_verbosity);
+
+        if isNonFailure(obj)
+            % skipped
+            % note: showing the reason is not supported
+            infix=sprintf('>\n  <%s />',outcome);
+        else
+            % XXX currently the entire error message is shown, together
+            % with the stack trace. It may be more convenient to show a
+            % summary of this information
+            message=moxunit_util_escape_xml(content.message);
+            stack_trace=moxunit_util_escape_xml(stack2str(content.stack));
+            infix=sprintf('>\n  <%s message="%s">%s</%s>',...
+                        outcome,message,stack_trace,outcome);
+        end
+
+        suffix=sprintf('\n</testcase>');
+    end
+
+    str=sprintf('%s%s%s',prefix,infix,suffix);
+
+function str=stack2str(stack)
+    n_stack=numel(stack);
+    lines=cell(1,n_stack);
+    for k=1:n_stack
+        s=stack(k);
+        lines{k}=sprintf('  %s:%d (%s)', ...
+                        s.name, s.line, s.file);
+    end
+    str=strjoin(lines,'\n');
+
+function s=get_classname(test_)
+    location=getLocation(test_);
+    test_name=getName(test_);
+    [pth,file_name]=fileparts(location);
+
+    s=sprintf('%s.%s',file_name,test_name);
```

Comment:
You don't want the test name to be part of the class name. The Junit XML viewer will put the name after the class name itself, so it is currently repeated.